File: src/modules/starship/starship-model.js
const mongoose = require("mongoose");

const starshipSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true,
  },
  model: {
    type: String,
  },
  manufacturer: {
    type: String,
  },
  cost_in_credits: {
    type: String,
    default: "10000000",
  },
  length: {
    type: String,
    default: "10000",
  },
  max_atmosphering_speed: {
    type: String,
  },
  crew: {
    type: String,
    default: "100",
  },
  passengers: {
    type: String,
    default: "100",
  },
  cargo_capacity: {
    type: String,
    default: "10000",
  },
  consumables: {
    type: String,
  },
  hyperdrive_rating: {
    type: Number,
    default: 2.0,
  },
  MGLT: {
    type: Number,
    default: 5,
  },
  starship_class: {
    type: String,
  },
  pilots: {
    type: [String],
  },
  films: {
    type: [String],
  },
  is_deleted: {
    type: Boolean,
  },
});

const Starship = mongoose.model("Starship", starshipSchema);

module.exports = Starship;

File: mongo-net-asp-csharp/modules/starship/starship-model.cs
using MongoDB.Bson;

namespace csharp_asp.modules.starship
{
    public class Starship : IStarship
    {
        public string _id { get; set; }
        public string Name { get; set; }
        public string? Model { get; set; }
        public string? Manufacturer { get; set; }
        public string? CostInCredits { get; set; }
        public string? Length { get; set; }
        public string? MaxAtmospheringSpeed { get; set; }
        public string? Crew { get; set; }
        public string? Passengers { get; set; }
        public string? CargoCapacity { get; set; }
        public string? Consumables { get; set; }
        public string? StarshipClass { get; set; }
        public int HyperdriveRating { get; set; }
        public int MGLT { get; set; }
        public List<string>? Pilots { get; set; }
        public List<string>? Films { get; set; }
        public Boolean IsDeleted { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        public Starship()
        {
            _id = ObjectId.GenerateNewId().ToString();
            Name = "NewStarship";
            Model = "Super";
            Manufacturer = "AgustinChavero";
            CostInCredits = "100000000000";
            Length = "400";
            MaxAtmospheringSpeed = "9999999999";
            Crew = "8";
            Passengers = "16";
            CargoCapacity = "40";
            Consumables = "All";
            StarshipClass = "Premium";
            HyperdriveRating = 9999999;
            MGLT = 30;
            IsDeleted = false;
        }
    }
}
